pkgs/by-name/au/autobrr/package.nix

```
@@ -0,0 +1,93 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  stdenvNoCC,
+  nix-update-script,
+  nodejs,
+  pnpm_9,
+  typescript,
+}:
+
+let
+  pname = "autobrr";
+  version = "1.56.1";
+  src = fetchFromGitHub {
+    owner = "autobrr";
+    repo = "autobrr";
+    tag = "v${version}";
+    hash = "sha256-2qo9UUaRvIqFEftHOH18GNzpRySvrNu1zl0vo+ZP3eU=";
+  };
+
+  autobrr-web = stdenvNoCC.mkDerivation {
+    pname = "${pname}-web";
+    inherit src version;
+
+    nativeBuildInputs = [
+      nodejs
+      pnpm_9.configHook
+      typescript
+    ];
+
+    sourceRoot = "${src.name}/web";
+
+    pnpmDeps = pnpm_9.fetchDeps {
+      inherit (autobrr-web)
+        pname
+        version
+        src
+        sourceRoot
+        ;
+      hash = "sha256-Iz2227+83RDuWlzR5zSBqVUplMl4vLb56Kef4nAH2vM=";
+    };
+
+    postBuild = ''
+      pnpm run build
+    '';
+
+    installPhase = ''
+      cp -r dist $out
+    '';
+  };
+in
+buildGoModule rec {
+  inherit
+    autobrr-web
+    pname
+    version
+    src
+    ;
+
+  vendorHash = "sha256-wsacy9Yr1gvDEzjIJJTq1Przr3zD2DLKzJwoyWpqHdM=";
+
+  preBuild = ''
+    cp -r ${autobrr-web}/* web/dist
+  '';
+
+  ldflags = [
+    "-X main.version=${version}"
+    "-X main.commit=${src.tag}"
+  ];
+
+  passthru.updateScript = nix-update-script {
```

An extra thing you could add is checking the version of one of the binaries:

```suggestion
  doInstallCheck = true;
  nativeInstallCheckInputs = [
    versionCheckHook
  ];
  versionCheckProgram = "${placeholder "out"}/bin/autobrrctl";
  versionCheckProgramArg = "version";

  passthru.updateScript = nix-update-script {
```

---

`autobrrctl version` checks for the most recent version and prints it after the current version. So I think `versionCheckHook` would always succeed, even if the actual version is incorrect.

---

> `autobrrctl version` checks for the most recent version and prints it after the current version. So I think `versionCheckHook` would always succeed, even if the actual version is incorrect.

In a regular build that's correct, but thankfully the nix sandbox disallows networking during most of the build time. From the output you can see the local version and it failing to fetch the version:

```console
autobrr> Executing versionCheckPhase                                                                                                                                                                              
autobrr> Successfully managed to find version 1.56.1 in the output of the command /nix/store/nmvccjllygw5l539gmcdra6il0nzb34c-autobrr-1.56.1/bin/autobrrctl version                                               
autobrr> Version: 1.56.1                                                                                                                                                                                          
autobrr> Commit: v1.56.1                                                                                                                                                                                          
autobrr> Build:                                                                                                                                                                                                   
autobrr> Failed to fetch latest release from api: Get "https://api.autobrr.com/repos/autobrr/autobrr/releases/latest": dial tcp: lookup api.autobrr.com on [::1]:53: read udp [::1]:58973->[::1]:53: read: connect
ion refused                                                                                                                                                                                                       
autobrr> Finished versionCheckPhase  
```